myokit/formats/sbml/_importer.py

```diff
@@ -37,332 +38,1016 @@ def __init__(self):
+                "The constraints will be ignored for the simulation.")
+
+        # Log warning if events are provided (could be supported in a later PR)
+        events = self._getListOfEvents(SBMLmodel)
+        if events:
             log.warn(
-                'The SBML version ' + str(sbml_version) + ' has not been'
-                ' tested. The model may not be imported correctly.')
+                "Myokit does not support SBML's events feature. The events"
+                " will be ignored for the simulation. Have a look at myokits"
+                " protocol feature for instantaneous state value changes.")
+
+        return model
+
+    def _getNamespace(self, element):
+        return split(element.tag)[0]
+
+    def _getModel(self, element):
+        model = element.find(
+            '{http://www.sbml.org/sbml/level3/version2/core}'
+            + 'model')
+        if model:
+            return model
+        return None
+
+    def _getName(self, element):
+        name = element.get('name')
+        if name:
+            return name
+        name = element.get('id')
+        if name:
+            return name
+        return None
+
+    def _getNotes(self, element):
+        notes = element.find(
+            '{http://www.sbml.org/sbml/level3/version2/core}'
+            + 'notes')
+        if notes:
+            return ET.tostring(notes).decode()
+        return None
+
+    def _getListOfFunctionDefinitions(self, element):
+         funcs = element.findall(
+            './'
+            + '{http://www.sbml.org/sbml/level3/version2/core}'
+            + 'listOfFunctionDefinitions/'
+            + '{http://www.sbml.org/sbml/level3/version2/core}'
+            + 'functionDefinition')
+        if funcs:
+            return funcs
+        return None
+
+    def _getListOfUnitDefinitions(self, element):
+        units = element.findall(
+            './'
+            + '{http://www.sbml.org/sbml/level3/version2/core}'
+            + 'listOfUnitDefinitions/'
+            + '{http://www.sbml.org/sbml/level3/version2/core}'
+            + 'unitDefinition')
+        if units:
+            return units
+        return None
+
+    def _getListOfCompartments(self, element):
+        comps = element.findall(
+            './'
+            + '{http://www.sbml.org/sbml/level3/version2/core}'
+            + 'listOfCompartments/'
+            + '{http://www.sbml.org/sbml/level3/version2/core}'
+            + 'compartment')
+        if comps:
+            return comps
+        return None
+
+    def _getListOfParameters(self, element):
+        params = element.findall(
+            './'
+            + '{http://www.sbml.org/sbml/level3/version2/core}'
+            + 'listOfParameters/'
+            + '{http://www.sbml.org/sbml/level3/version2/core}'
+            + 'parameter')
+        if params:
+            return params
+        return None
+
+    def _getListOfSpecies(self, element):
```

Comment:
How do these work? Can there be more than one "listOfSpecies" in the document?

Reply:
No, I don't think so. As far as, I understood are models structured like this

beginning of model definition
list of function definitions (optional) (Section 4.3)
list of unit definitions (optional) (Section 4.4)
list of compartments (optional) (Section 4.5)
list of species (optional) (Section 4.6)
list of parameters (optional) (Section 4.7)
list of initial assignments (optional) (Section 4.8)
list of rules (optional) (Section 4.9)
list of constraints (optional) (Section 4.10)
list of reactions (optional) (Section 4.11)
list of events (optional) (Section 4.12)
end of model definition

where each list only appears once (although I don't remember reading this explicitly somewhere).

But since each list has an arbitrary number of elements, i.e.

listOfSpecies
     species
     species
listOfSpecies

there wouldn't be a need to have multiple list objects.